### PR TITLE
Add roadmap page

### PR DESCRIPTION
**Pull request by meiliang86**
**What changed?**
Adding a page to talk about cadence roadmap.

**Why?**
Helps the community to understand what has been worked on and what is in future plan.

**How did you test it?**
n/a

**Potential risks**
n/a

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+* Service availability and reliability improvements
+* [Domain level AuthN and AuthZ support](https://github.com/uber/cadence/issues/2833)
+* Graceful domain failover design
+* Multi-tenancy improvements and task processing redesign
+* UI bug fixes and performance improvements
+
+## Q2 2020
+
+* Kafka deprecation for Cadence replication
+* Graceful domain failover
```
Comment (wxing1292) [inline docs/roadmap.md:33]: Best effort graceful domain failover?

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* Multi-tanancy improvements and task processing
```
Comment (wxing1292) [inline docs/roadmap.md:34]: `task processing` -> `task execution prioritization and isolation`?

File: docs/roadmap.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+* MySQL support
+* [Visibility on ElasticSearch](https://github.com/uber/cadence/blob/master/docs/visibility-on-elasticsearch.md)
+
+## Q4 2019
+
+* [Multi-DC support for Cadence replication](https://github.com/uber/cadence/blob/master/docs/design/2290-cadence-ndc.md)
+* Workflow history archival
+* Workflow visibility archival
+* [Synchronous Request Reply](https://github.com/uber/cadence/blob/master/docs/design/2215-synchronous-request-reply.md)
+* Postgres SQL support
```
Comment (wxing1292) [inline docs/roadmap.md:20]: this one is not prod ready

File: docs/roadmap.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+* Graceful domain failover
+* Multi-tanancy improvements and task processing
+* UI and CLI feature parity
+* GRPC support
+
+## Q3 2020 and beyond
+
+* Parent-child affinity
+* Task priority
+* Reset 2.0 (more flexibility and handle child)
```
Comment (wxing1292) [inline docs/roadmap.md:42]: child workflow can be handled the same way as NDC child workflow conflict resolution.
so once the create & terminate workflow is done, this should be easy

File: docs/roadmap.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Cadence Roadmap
+
+The following is a high-level quarterly roadmap of the [Cadence](https://cadenceworkflow.io/) project.
+Please contact [Cadence discussion group](https://groups.google.com/d/forum/cadence-discussion) for more information.
+
+## Q3 2019
+
+* [Resource-Specific Tasklist (a.k.a session)](https://github.com/uber/cadence/blob/master/docs/design/1533-host-specific-tasklist.md)
+* [Visibility on Elastic Search](https://github.com/uber/cadence/blob/master/docs/visibility-on-elasticsearch.md)
```
Comment (vancexu) [inline docs/roadmap.md:9]: remove since it is duplicate?

File: docs/roadmap.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+* Multi-tenancy improvements and task processing redesign
+* UI bug fixes and performance improvements
+
+## Q2 2020
+
+* Kafka deprecation for Cadence replication
+* Graceful domain failover
+* Multi-tanancy improvements and task processing
+* UI and CLI feature parity
+* GRPC support
```
Comment (vancexu) [inline docs/roadmap.md:36]: gRPC design?

**Pull request merged**